papyri/gen.py

```diff
@@ -1202,146 +1293,62 @@ def get_example_data(
-                                    figs.append((figname, fig))
-                                    print_(
-                                        f"Still fig manager(s) open for {qa}: {figname}"
-                                    )
-                                plt.close("all")
-                            fig_managers = executor.fig_man()
-                            assert len(fig_managers) == 0, fig_managers + [
-                                did_except,
-                            ]
-                    # we've executed, we now want to compare output
-                    # in the docstring with the one we produced.
-                    if (out == repr(res)) or (res is None and out == []):
-                        pass
-                    else:
-                        pass
-                        # captured output differ TBD
-                entries = parse_script(script, ns=ns, prev=acc, config=config, where=qa)
-                if entries is None:
-                    entries = [("jedi failed", "jedi failed")]
-                entries = _add_classes(entries)
-                tok_entries = [GenToken(*x) for x in entries]  # type: ignore
 
-                acc += "\n" + script
-                example_section_data.append(
-                    Code(tok_entries, "\n".join(item.out), ce_status)
-                )
-                for figname, _ in figs:
-                    example_section_data.append(
-                        Fig(
-                            RefInfo.from_untrusted(
-                                self.root, self.version, "assets", figname
-                            )
-                        )
-                    )
-                all_figs.extend(figs)
+        sys_stdout = sys.stdout
+
+        def dbg(*args):
+            for arg in args:
+                sys_stdout.write(f"{arg}\n")
+            sys_stdout.flush()
+
+        try:
+            filename = inspect.getfile(obj)
+        except TypeError:
+            filename = None
+        try:
+            lineno = inspect.getsourcelines(obj)[1]
+        except (TypeError, OSError):
+            lineno = None
+
+        doctest_runner = PapyriDocTestRunner(
+            gen=self,
+            obj=obj,
+            qa=qa,
+            config=config,
+            # TODO: Make optionflags configurable
+            optionflags=doctest.ELLIPSIS,
+        )
+        example_section_data = Section([], None)
+
+        def debugprint(*args):
+            """
+            version of print that capture current stdout to use during testing to debug
+            """
+            sys_stdout.write(" ".join(str(x) for x in args) + "\n")
 
-        # TODO fix this if plt.close not called and still a ligering figure.
-        fig_managers = executor.fig_man()
+        for block in blocks:
+            doctests = doctest.DocTestParser().get_doctest(
+                block, doctest_runner.globs, obj.__name__, filename, lineno
+            )
```

Comment:
get_doctests also completely drop interleaving text, so we'll need to fallback to `parse(...)` and create the DocTests object ourselves here.